core/src/main/java/com/linecorp/armeria/client/RefreshingAddressResolver.java

```diff
@@ -220,8 +221,7 @@ public void onRemoval(DnsQuestion question, @Nullable List<DnsRecord> records,
         final CacheEntry entry = addressResolverCache.getIfPresent(hostname);
         if (entry != null) {
             if (entry.refreshable()) {
-                // onRemoval is invoked by the executor of 'dnsResolverCache'.
-                executor().execute(entry::refresh);
```

Comment:
refresh should be called by the entry's executor not by the executor who gets this cache entry from `addressResolverCache`.